meson.build

```diff
@@ -659,55 +659,80 @@ endif
 
 ###############################################################
 # Library: Readline
-#
-# FIXME: editline support
 ###############################################################
 
 if not get_option('readline').disabled()
-  readline = dependency('readline', required: false)
-  if not readline.found()
-    readline = cc.find_library('readline',
-      required: get_option('readline').enabled(),
-      dirs: g_c_lib)
-  endif
+  libedit_preferred = get_option('libedit_preferred')
+  # Set the order of readline dependencies
+  check_readline_deps = libedit_preferred ? \
+    ['libedit', 'readline'] : ['readline', 'libedit']
+
+  foreach readline_dep : check_readline_deps
+    readline = dependency(readline_dep, required: false)
+    if not readline.found()
+      readline = cc.find_library(readline_dep,
+        required: get_option('readline').enabled(),
+        dirs: g_c_lib)
+    endif
+    if readline.found()
+      break
+    endif
+  endforeach
 
   if readline.found()
     cdata.set('HAVE_LIBREADLINE', 1)
 
-    if cc.has_header('readline/history.h',
-        args: g_c_args, include_directories: g_c_inc,
-        dependencies: [readline], required: false)
-      history_h = 'readline/history.h'
-      cdata.set('HAVE_READLINE_HISTORY_H', 1)
-      cdata.set('HAVE_READLINE_H', false)
-    elif cc.has_header('history.h',
-        args: g_c_args, include_directories: g_c_inc,
-        dependencies: [readline], required: false)
-      history_h = 'history.h'
-      cdata.set('HAVE_READLINE_HISTORY_H', false)
-      cdata.set('HAVE_HISTORY_H', 1)
-    else
-      error('''readline header not found
-If you have readline already installed, see see meson-log/meson-log.txt for details on the
-failure.  It is possible the compiler isn't looking in the proper directory.
-Use -Dreadline=false to disable readline support.''')
-    endif
+    libedit_flags = {
+      'history_h': 'editline/history.h',
+      'history_h_flag': 'HAVE_EDITLINE_HISTORY_H',
+      'readline_h': 'editline/readline.h',
+      'readline_h_flag': 'HAVE_EDITLINE_READLINE_H'
+    }
+    readline_flags = {
+      'history_h': 'readline/history.h',
+      'history_h_flag': 'HAVE_READLINE_HISTORY_H',
+      'readline_h': 'readline/readline.h',
+      'readline_h_flag': 'HAVE_READLINE_READLINE_H'
+    }
+    raw_flags = {
```

Comment:
What does "raw" mean here?

Reply:
I could not find alternative name. Maybe "default" or any suggestions?

Reply:
I'm not even quite sure what you're trying to do - it's just the check for readline installed not inside readline/?

Reply:
Let me try to explain.

There are two available options, user will prefer:

1. libedit over readline (if `libedit_preferred` is true)
2. readline over libedit (default)

In autoconf configuration, if libedit is preferred; checking for `history.h and readline.h` order is

1. editline
2. without editline and readline
3. readline


if readline is preferred, order is

1. readline
2. without editline and readline
3. editline

So, I created these variables namely `libedit_flags, readline_flags and raw_flags(which is for without editline and readline)`.
And I set the order of these variables based on `libedit_preferred` meson option for checking these `history.h and readline.h`.

Does that look good or do you prefer something else?